server/test/unit/BraveAlerterConfiguratorTest/alertSessionChangedCallbackTest.js

```diff
@@ -64,18 +65,67 @@ describe('BraveAlerterConfigurator.js unit tests: alertSessionChangedCallback',
     const braveAlerter = braveAlerterConfigurator.createBraveAlerter()
     await braveAlerter.alertSessionChangedCallback(new AlertSession(sessionId, CHATBOT_STATE.WAITING_FOR_CATEGORY, newRespondedByPhoneNumber))
 
-    const expectedSession = sessionFactory(
-      sessionFactory({
-        id: sessionId,
-        chatbotState: CHATBOT_STATE.WAITING_FOR_CATEGORY,
-        respondedAt: this.testCurrentTime,
-        respondedByPhoneNumber: newRespondedByPhoneNumber,
-      }),
-    )
+    const expectedSession = sessionFactory({
+      id: sessionId,
+      chatbotState: CHATBOT_STATE.WAITING_FOR_CATEGORY,
+      respondedAt: this.testCurrentTime,
+      respondedByPhoneNumber: newRespondedByPhoneNumber,
+    })
 
     expect(db.saveSession).to.be.calledWith(expectedSession, sandbox.any)
   })
 
+  it('if given alertState WAITING_FOR_CATEGORY and it has not already been responded to should reset the stillness timer in the firmware state machine', async () => {
+    const sessionId = 'ca6e85b1-0a8c-4e1a-8d1e-7a35f838d7bc'
+    const newRespondedByPhoneNumber = '+18887774444'
+    sandbox.stub(db, 'getSessionWithSessionId').returns(sessionFactory({ id: sessionId, respondedAt: null, respondedByPhoneNumber: null }))
+    sandbox.stub(particleFunctions, 'resetStillnessTimer')
+
+    const braveAlerterConfigurator = new BraveAlerterConfigurator()
+    const braveAlerter = braveAlerterConfigurator.createBraveAlerter()
+    await braveAlerter.alertSessionChangedCallback(new AlertSession(sessionId, CHATBOT_STATE.WAITING_FOR_CATEGORY, newRespondedByPhoneNumber))
+
+    const expectedSession = sessionFactory({
+      id: sessionId,
+      chatbotState: CHATBOT_STATE.WAITING_FOR_CATEGORY,
+      respondedAt: this.testCurrentTime,
+      respondedByPhoneNumber: newRespondedByPhoneNumber,
+    })
+
+    expect(particleFunctions.resetStillnessTimer).to.be.calledWith(expectedSession.location.radarCoreId, helpers.getEnvVar('PARTICLE_PRODUCT_GROUP'))
+  })
```

Comment:
I'm not as clear on what this is testing. Is it whether the function is being called with the appropriate arguments?

Reply:
This is testing the condition that we actually call the Particle Cloud function to reset the Stillness Timer.

The `alertSessionChangedCallback` is a function that is called by Brave Alert Lib whenever there is a chatbot interaction. As a parameter, it sends the and `AlertSession` object containing information about what the new state of the session is. Then the `alertSessionChangedCallback` function looks at the given `AlertSession` object to figure out what it wants to do.

What we really want to do is call `particleFunctions.resetStillnessTimer` when the responder texts us "OK". But the way we have to do that is by looking at the properties of the `AlertSession` object that is passed to `alertSessionChangedCallback`.

When a Responder texts "OK" we are either moving from `CHATBOT_STATE.STARTED` or `CHATBOT_STATE.WAITING_FOR_REPLY` to `CHATBOT_STATE.WAITING_FOR_CATEGORY`. And the `AlertSession` object always contains the new properties, not the old ones. So our trigger will be when it sends us `CHATBOT_STATE_WAITING_FOR_CATEGORY`. And we only want to reset the stillness timer the *first* time that the session is in `WAITING_FOR_CATEGORY`. 

However, it is possible to transition to `WAITING_FOR_CATEGORY` more than once in the same session by responding with an invalid number when asked to categorize the alert. So we look at the Session object returned from the DB to see if it already had a `respondedAt` or `respondedByPhoneNumber` field. If these are `null` or `undefined` then we know that this is the first time that this session is in the `WAITING_FOR_CATEGORY` state.

This is kind of what I was saying about how using Brave Alert Lib ended up making things more complicated than they necessarily needed to be. Please let me know if this still doesn't make sense!

Reply:
Thanks for explaining Theresa! I think I understand now. 

So your test is making sure that `particleFunctions.resetStillnessTimer` is getting called when the chatbot_state is changed to `CHATBOT_STATE_WAITING_FOR_CATEGORY` for the _first_ time.

Reply:
Exactly!